src/main/java/com/minecolonies/core/colony/managers/TravellingManager.java

```diff
@@ -0,0 +1,216 @@
+package com.minecolonies.core.colony.managers;
+
+import com.google.common.collect.Maps;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.managers.interfaces.ITravellingManager;
+import com.minecolonies.api.entity.citizen.AbstractEntityCitizen;
+import com.minecolonies.api.util.EntityUtils;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.ColonyConstants;
+import com.minecolonies.api.util.constant.NbtTagConstants;
+import com.minecolonies.core.util.TeleportHelper;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.NbtUtils;
+import net.minecraft.nbt.Tag;
+import net.minecraftforge.common.util.INBTSerializable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class TravellingManager implements ITravellingManager, INBTSerializable<CompoundTag>
+{
+
+    private final IColony                    colony;
+    private final Map<Integer, TravelerData> travelerDataMap = Maps.newHashMap();
+
+    public TravellingManager(final IColony colony) {this.colony = colony;}
+
+    @Override
+    public boolean isTravelling(final int citizenId)
+    {
+        return Optional.ofNullable(travelerDataMap.get(citizenId)).map(TravelerData::isTraveling).orElse(false);
+    }
+
+    @Override
+    public Optional<BlockPos> getTravellingTargetFor(final int citizenId)
+    {
+        return Optional.ofNullable(travelerDataMap.get(citizenId)).map(TravelerData::getTarget);
+    }
+
+    @Override
+    public void startTravellingTo(final int citizenId, final BlockPos target, final int travelTimeInTicks)
+    {
+        travelerDataMap.put(citizenId, new TravelerData(citizenId, target, travelTimeInTicks));
+        colony.markDirty();
+    }
+
+    @Override
+    public void finishTravellingFor(final int citizenId)
+    {
+        if (travelerDataMap.containsKey(citizenId))
+        {
+            travelerDataMap.remove(citizenId);
+            colony.markDirty();
+        }
+    }
+
+    @Override
+    public void recallAllTravellingCitizens()
+    {
+        for (final Integer citizenId : travelerDataMap.keySet())
+        {
+            final ICitizenData citizenData = this.colony.getCitizenManager().getCivilian(citizenId);
```

Comment:
does the citizen still exists if it somehow dies during the travel?  I recall the nether worker can take damage and die. If not probably need a null check here

Reply:
ah then it could only crash on loading backups via cmd